test-tests/shared/src/test/scala/zio/test/AssertionSpec.scala

```diff
@@ -72,6 +72,36 @@ object AssertionSpec
         test("hasSize must succeed when iterable size is equal to specified assertion") {
           assert(Seq(1, 2, 3), hasSize(equalTo(3)))
         },
+        test("hasFirst must fail when an iterable is empty") {
+          assert(Seq(), hasFirst(anything))
+        } @@ failure,
+        test("hasFirst must succeed when a head is equal to a specific assertion") {
+          assert(Seq(1, 2, 3), hasFirst(equalTo(1)))
+        },
+        test("hasFirst must fail when a head is not equal to a specific assertion") {
+          assert(Seq(1, 2, 3), hasFirst(equalTo(100)))
+        } @@ failure,
+        test("hasLast must fail when an iterable is empty") {
+          assert(Seq(), hasLast(anything))
+        } @@ failure,
+        test("hasLast must succeed when a last is equal to a specific assertion") {
+          assert(Seq(1, 2, 3), hasLast(equalTo(3)))
+        },
+        test("hasLast must fail when a last is not equal to specific assertion") {
+          assert(Seq(1, 2, 3), hasLast(equalTo(100)))
+        } @@ failure,
+        test("hasAt must fail when an idnex is outside of a sequence range") {
+          assert(Seq(1, 2, 3), hasAt(-1)(anything))
+        } @@ failure,
+        test("hasAt must fail when an idnex is outside of a sequence range 2") {
+          assert(Seq(1, 2, 3), hasAt(3)(anything))
+        } @@ failure,
+        test("hasAt must fail when head is not equal to a specific assertion") {
+          assert(Seq(1, 2, 3), hasAt(1)(equalTo(100)))
+        } @@ failure,
+        test("hasAt must succeed when a head is equal to a specific assertion") {
```

Comment:
You want to have at least one test that succeeds, right?

Reply:
Right! I created a new one